doc/tutorial.md

```diff
@@ -0,0 +1,607 @@
+platform_main_CFLAGS = -I${top_srcdir}/libubx  @UBX_CFLAGS@
+platform_main_LDFLAGS = -module -avoid-version -shared -export-dynamic  @UBX_LIBS@ -ldl
+```
+
+Here we specifiy that the name of the executable is _platform_main_
+It might be possible that, if some custom types are used in the configuration, but are not installed, they mist be added to the _CFLAGS_:
+
+```cmake
+platform_main_CFLAGS = -I${top_srcdir}/libubx -I path/to/other/headers  @UBX_CFLAGS@
+```
+
+In order to compile, we will use the same commands as before (we do not need to install).
+```bash
+autoreconf --install
+./configure
+make
+```
+
+## The program
+The main follows the same structure of the .usc file.
+### Libraries
+It start with some include (struct that are needed in configuration) and loading the libraries
+```c
+#include <ubx.h>
+
+#define WEBIF_PORT "8810"
+#define DOUBLE_STR "double"
+#include "ptrig_config.h"
+#include "ptrig_period.h"
+int main()
+{
+  int len, ret=EXIT_FAILURE;
+  ubx_node_info_t ni;
+  ubx_block_t *plat1, *control1, *logger1, *ptrig1, *webif, *fifo_vel, *fifo_pos;
+  ubx_data_t *d;
+
+  /* initalize the node */
+  ubx_node_init(&ni, "platform_and_control");
+const char *modules[8];
+modules[0]= "/usr/local/lib/ubx/0.6/stdtypes.so";
+modules[1]= "/usr/local/lib/ubx/0.6/stattypes.so";
+modules[2]= "/usr/local/lib/ubx/0.6/ptrig.so";
+modules[3]= "/usr/local/lib/ubx/0.6/platform_2dof.so";
+modules[4]= "/usr/local/lib/ubx/0.6/platform_2dof_control.so";
+modules[5]= "/usr/local/lib/ubx/0.6/webif.so";
+modules[6]= "/usr/local/lib/ubx/0.6/logger.so";
+modules[7]= "/usr/local/lib/ubx/0.6/lfds_cyclic.so";
+  /* load modules */
+for (int i=0; i<8;i++)
+  if(ubx_module_load(&ni, modules[i]) != 0){
+      printf("fail to load %s",modules[i]);
+      goto out;
+    }
+```
+### Block instantiation
+Then, we instantiate blocks (code for only one, for sake of brevity):
+
+```c
+if((plat1 = ubx_block_create(&ni, "platform_2dof", "plat1"))==NULL){
+      printf("fail to create platform_2dof");
+      goto out;
+    }
+```
+
+### Property configuration
+Now we have the more tedious and error-prone part, that is the configuration. depending by the type of the property, it might be necessary to allocate memory with the function `ubx_data_resize`, that takes as argument the data pointer, and the new length.
+#### String property:
+```c
+d = ubx_config_get_data(webif, "port");
+len = strlen(WEBIF_PORT)+1;
+/* resize the char array as necessary and copy the port string */
+ubx_data_resize(d, len);
+strncpy((char *)d->data, WEBIF_PORT, len);
+```
+Here the sting is declared with a `#define`, it can be written directly, or with a variable, _e.g._ `char filename[]="/tmp/platform_time.log"`; 
+
+#### Double property:
+```
+d = ubx_config_get_data(control1, "gain");
+*((double*)d->data)=0.12;
+```
+  In this case, no memory allocation is necessary. The second line says: consider `d->data` as  a pointer to double, and assign to the pointed memory area the value `0.12`.
```

Comment:
The allocation is necessary here too! It happens to work because by chance there is some memory there, but you musn't rely on it.
So you should resize it to as above to length `1`